server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSource.java

```diff
@@ -237,6 +237,66 @@ public boolean needsScores() {
                 return script.needs_score();
             }
         }
+
+        // No need to implement ReaderContextAware here, the delegate already takes care of updating data structures
+        public static class WithScript extends Bytes {
```

Comment:
Huh, was this just not implemented before, or used a generic implementation or something?  The deep hierarchy of this file makes it hard to understand how everything fits together :(

Reply:
Oh, I think I see... it was a "top-level" item class and not consistent with how the others were nested?

Reply:
Correct.  It was top level, but behaved as if it was a sub-type of Bytes.  Moving it here lets us say "every top level ValuesSource should have a corresponding ValuesSourceType implementation, which then chooses the correct sub-class".  There's still some rough edges around that, but it's a pretty good way to think about it.

Reply:
:+1: